src/mygrad/nnet/initializers/he_uniform.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+from mygrad import Tensor
+
+
+def he_uniform(*shape, gain=1, dtype=np.float32, constant=False):
+    """ Initialize a ``mygrad.Tensor`` according to the uniform initialization procedure
+    described by He et al.
+
+    Parameters
+    ----------
+    shape : Sequence[int]
+        The shape of the output Tensor. Note that ``shape`` must be at least two-dimensional.
+
+    gain : Real, optional (default=1)
+        The gain (scaling factor) to apply.
+
+    dtype : data-type, optional (default=float32)
+        The data type of the output tensor; must be a floating-point type.
+
+    constant : bool, optional (default=False)
+        If `True`, the returned tensor is a constant (it
+            does not back-propagate a gradient).
+
+    Returns
+    -------
+    mygrad.Tensor, shape=``shape``
+        A Tensor, with values initialized according to the He uniform initialization.
+
+    Extended Description
```

Comment:
This is great. I would really like to go through and add "takeaways" to these sections. I.e. "this paper recommends using this initialization for X circumstances; it makes such and such assumptions"

These are things that I end up having to re-read the paper for and wish they were just documented. This need not be done for this round of PRs, but it is something for us to keep in mind. I also think that will help make apparent the value of `mygrad` as an educational tool :)

Reply:
That's a great addition; definitely more informative than just having the paper linked